src/containerapp/azext_containerapp/_params.py

```diff
@@ -0,0 +1,174 @@
+                                                get_resource_name_completion_list, file_type,
+                                                get_three_state_flag, get_enum_type, tags_type)
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+
+from ._validators import (validate_memory, validate_cpu, validate_managed_env_name_or_id, validate_registry_server,
+                          validate_registry_user, validate_registry_pass, validate_target_port, validate_ingress)
+
+def load_arguments(self, _):
+
+    name_type = CLIArgumentType(options_list=['--name', '-n'])
+
+    with self.argument_context('containerapp') as c:
+        # Base arguments
+        c.argument('name', name_type, metavar='NAME', id_part='name')
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+
+    with self.argument_context('containerapp') as c:
+        c.argument('tags', arg_type=tags_type)
+        c.argument('managed_env', validator=validate_managed_env_name_or_id, options_list=['--environment'], help="Name or resource ID of the container app's environment.")
+        c.argument('yaml', type=file_type, help='Path to a .yaml file with the configuration of a container app. All other parameters will be ignored. For an example, see  https://docs.microsoft.com/azure/container-apps/azure-resource-manager-api-spec#examples')
+
+    # Container
+    with self.argument_context('containerapp', arg_group='Container (Creates new revision)') as c:
+        c.argument('image', type=str, options_list=['--image', '-i'], help="Container image, e.g. publisher/image-name:tag.")
+        c.argument('container_name', type=str, options_list=['--container-name'], help="Name of the container.")
+        c.argument('cpu', type=float, validator=validate_cpu, options_list=['--cpu'], help="Required CPU in cores, e.g. 0.5")
+        c.argument('memory', type=str, validator=validate_memory, options_list=['--memory'], help="Required memory, e.g. 1.0Gi")
+        c.argument('env_vars', nargs='*', options_list=['--env-vars'], help="A list of environment variable(s) for the container. Space-separated values in 'key=value' format. Empty string to clear existing values. Prefix value with 'secretref:' to reference a secret.")
+        c.argument('startup_command', nargs='*', options_list=['--command'], help="A list of supported commands on the container that will executed during startup. Space-separated values e.g. \"/bin/queue\" \"mycommand\". Empty string to clear existing values")
+        c.argument('args', nargs='*', options_list=['--args'], help="A list of container startup command argument(s). Space-separated values e.g. \"-c\" \"mycommand\". Empty string to clear existing values")
+        c.argument('revision_suffix', type=str, options_list=['--revision-suffix'], help='User friendly suffix that is appended to the revision name')
```

Comment:
The default value of these arguments is the argument name, so `options_list` does not need to be defined repeatedly